project2/py/web_scraper.py

```diff
@@ -45,6 +46,42 @@ def parse_ingredients(ingredients):
     objs = utils.pmap(Ingredient, ingredients)
     return objs
 
+def parse_methods(directions, kb):
+    allMethods = []
+    for direction in directions:
+        direction = direction.lower().replace('.,', '')
+        allMethods.extend(extract_methods(direction, kb))
+
+    allMethods = list(set(allMethods))
+    return allMethods
+
+def parse_tools(directions, kb):
+    allTools = []
+    for direction in directions:
+        direction = direction.lower().replace('.,', '')
+        allTools.extend(extract_tools(direction, kb))
+
+    allTools = list(set(allTools))
+    return allTools
+
+def extract_methods(direction, kb):
+    methods = []
+    for word in kb.methods:
+        if word in direction:
+            methods.append(word.capitalize())
+    return methods
+
+def extract_tools(direction, kb):
+    tools = []
+    for word in kb.tools:
+        if word in direction:
+            tools.append(word.capitalize())
+    return tools
+
+def remove_dupes(str):
```

Comment:
no `set` ? 😢

Reply:
i'm not super familiar with how to use set, so i added them to a list and then use set in parse_tools and parse_methods to take advantage of its ability to remove duplicates

i can change if necessary, but may also need to ask some questions in msngr